src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write code here
+const sortEmployeesBySalary = function (selector) {
+  // Получаем все элементы списка
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив и сортируем его по зарплате
+  const sortedEmployees = Array.from(employeeItems).sort((a, b) => {
+    // Извлекаем зарплату из атрибутов данных и преобразуем в число
+    const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+    // Удаляем $ и запятые
+    const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+    return salaryB - salaryA; // Сортируем по убыванию
+  });
+
+  // Очищаем существующий список и добавляем отсортированные элементы
+  const employeeList = document.querySelector('ul');
```

Comment:
Code Quality Issue: You are querying the DOM for 'ul' which is a very generic selector. This could lead to unexpected results if there are multiple 'ul' elements on the page. It would be better to use a more specific selector or pass the parent element as a parameter to the function.